solr/app.py

```diff
@@ -17,10 +19,37 @@ def create_app():
     app.url_map.strict_slashes = False
     Consul(app)  # load_config expects consul to be registered
     load_config(app)
+    if config:
+        app.config.update(config)
+        
+    db.init_app(app)
     logging.config.dictConfig(
         app.config['SOLR_SERVICE_LOGGING']
     )
     
+    ## pysqlite driver breaks transactions, we have to apply some hacks as per
```

Comment:
My first point was tied to the usage of the SQLite driver monkey patch. I think it's only relevant to place it here if we are expecting to use that driver in production. As it stands, I don't think we intend to use SQLite, but to use postgres.

If we were to use postgres, I would even go as far as to include it in the tests, and hence `testing.postgres` -- this is of course out of personal preference. And so at this stage, we could remove in-memory SQL database, and so remove the monkey patch.

As per the example I gave, there is some postgres dialect in several methods down, but I wasn't referring to that. I simply meant use `testing.postgres` to create a dummy postgres instance, in which you can interact with it via the ORM -- this requires no monkey patching (or postgres dialect).

I feel it is more explicit saying: I expect to use postgres, and my models are written with that in mind (and are deployed to production like so), and so I am testing against that in my unit/functional tests. However, it doesn't mean if I wanted to switch to SQLite, I couldn't, and could include any monkey patches relevant for that driver if it needed it.